pkg/config/config.go

```diff
@@ -40,11 +40,20 @@ func New(viper *viper.Viper) (*Config, error) {
 		Timeout: 10 * time.Second,
 	}
 
-	apiClient, err := api.NewClient(&api.Config{Address: viper.GetString("VAULT_ADDR"), HttpClient: httpClient})
+	apiConfig := &api.Config{
+		Address:    viper.GetString("VAULT_ADDR"),
+		HttpClient: httpClient,
+	}
+
+	apiClient, err := api.NewClient(apiConfig)
 	if err != nil {
 		return nil, err
 	}
 
+	if viper.IsSet("VAULT_NAMESPACE") {
+		apiClient.SetNamespace(viper.GetString("VAULT_NAMESPACE"))
```

Comment:
This looks good - I skimmed the namespace docs; it seems this will work even when the PATH_PREFIX or `avp_path` is set to something namespace agnostic, like `secret/data/my-app` right?

Reply:
Right, it should basically prepend the namespace to the path it tries to call.
`vault_addr/namespace/prefix | avp_path`